src/app/editor.js

```diff
@@ -100,9 +110,15 @@ function Editor (loadingFromGist) {
   };
 
   this.clearAnnotations = function () {
+    sourceAnnotations = [];
     editor.getSession().clearAnnotations();
   };
 
+  this.addAnnotation = function (annotation) {
+    sourceAnnotations[sourceAnnotations.length] = annotation;
+    this.setAnnotations(sourceAnnotations);
+  };
+
   this.setAnnotations = function (sourceAnnotations) {
     editor.getSession().setAnnotations(sourceAnnotations);
```

Comment:
Oh I'm sorry. Of course the array does not have to be updated here. But nevertheless, I think it is good practice to have only one entry point into updating the `sourceAnnotations` array which then also calls `editor.getSennio().setAnnotations`.